admin/commands/pmm_server_upgrade/start/start.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package start holds logic for starting pmm-server-upgrade.
+package start
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_validator "github.com/grpc-ecosystem/go-grpc-middleware/validator"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	channelz "google.golang.org/grpc/channelz/service"
+	"google.golang.org/grpc/reflection"
+
+	"github.com/percona/pmm/admin/cli/flags"
+	"github.com/percona/pmm/admin/commands"
+	"github.com/percona/pmm/admin/pkg/docker"
+	"github.com/percona/pmm/admin/services/status"
+	"github.com/percona/pmm/admin/services/update"
+	"github.com/percona/pmm/api/updatepb"
+)
+
+const (
+	gRPCMessageMaxSize = 100 * 1024 * 1024
+	socketPath         = "/srv/pmm-server-upgrade.sock"
+	shutdownTimeout    = 1 * time.Second
+)
+
+// StartCommand is used by Kong for CLI flags and commands.
+type StartCommand struct {
+	DockerImage string `default:"percona/pmm-server:2" help:"Docker image to use for updating to the latest version"`
+
+	dockerFn containerManager
+	globals  *flags.GlobalFlags
+}
+
+type startResult struct{}
+
+// Result is a command run result.
+func (r *startResult) Result() {}
+
+// String stringifies command result.
+func (r *startResult) String() string {
+	return "Exiting"
```

Comment:
Yes and no. I think it makes sense here because this is printed when the `start` command finishes which stops the daemon and exits
But there's still a lot of work to be done with the UI/UX in general. I'd like to address it all at the same time.

Reply:
shouldn't then the `start` command be renamed to `run` if it stops the daemon once it's finished?

Reply:
Happy to rename if `run` is more appropriate.
I'd prefer `start` or `daemon` but it's just a personal preference.

Reply:
I think renaming to `run` makes sense, in the case of `start` or `daemon` we should just start the daemon in the background and return the result to the process which invoked this command.